modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/DashboardAnonViewTest.java

```diff
@@ -288,24 +186,33 @@ public void testAnonDashboardPageRemove() throws Exception {
     @Test(groups = "wso2.ds.dashboard", description = "Remove anonymous view mode form dashboard",
             dependsOnMethods = "testAnonDashboardPageRemove")
     public void testRemoveAnonModeFromDashboard() throws Exception {
-        String defaultViewGadgetExpected = "USA Map";
-
-        getDriver().get(getBaseUrl() + "/portal/dashboards/" + dashboardTitle + "?editor=true");
-        getDriver().findElement(By.cssSelector(".ues-page-properties-toggle")).click();
+        redirectToLocation("portal", "dashboards/" + dashboardTitle + "?editor=true");
+        getDriver().findElement(By.cssSelector("a.ues-pages-toggle")).click();
+        switchPage("landing");
         getDriver().findElement(By.cssSelector("input[name='anon']")).click();
-
         boolean isToggleButtonHidden = getDriver().findElements(By.cssSelector(".toggle-design-view.hide")).size() > 0;
-        String defaultGadgetActual = getDriver().findElement(By.cssSelector("iframe")).getAttribute("title");
-        getDriver().get(getBaseUrl() + "/portal/dashboards/" + dashboardTitle + "/landing?isAnonView=true");
-        boolean noIFramesAvailable = getDriver().findElements(By.cssSelector("iframe")).size() <= 0;
-
-        getDriver().close();
 
-        assertEquals(defaultViewGadgetExpected, defaultGadgetActual);
         assertTrue(isToggleButtonHidden, "Anonymous toggle button is not hidden");
+        assertEquals(GADGET_1, getAttributeValue("iframe", "title"));
+
+        redirectToLocation("portal", "dashboards/" + dashboardTitle + "/landing?isAnonView=true");
+        modifyTimeOut(2);
+        boolean noIFramesAvailable = getDriver().findElements(By.cssSelector("iframe")).size() <= 0;
+        resetTimeOut();
         assertTrue(noIFramesAvailable, "There are gadgets available for anonymous view when no anonymous view available");
     }
 
+    /**
+     * Get the value of given attribute of given element
+     *
+     * @param element       name of the element
+     * @param attributeName name of the attribute
+     * @return {String}
+     */
+    public String getAttributeValue(String element, String attributeName) throws Exception {
```

Comment:
Why it's public? If this method needs to be used within this class then it should be a private method.